File: RecursiveFunctionJavaSample/app/src/main/java/com/template/recursivefunctionjavasample/MainActivity.java
package com.template.recursivefunctionjavasample;

import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ViewUtil.setOnClickListenerForToggleButton(findViewById(R.id.content), v -> {
            final TextView view = (TextView) v;
            Toast.makeText(MainActivity.this, view.getText(), Toast.LENGTH_SHORT).show();
        });
    }
}

File: RecursiveFunctionKotlinSample/app/src/main/java/com/template/recursivefunctionkotlinsample/MainActivity.kt
package com.template.recursivefunctionkotlinsample

import android.os.Bundle
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ViewUtil.setOnClickListenerForToggleButton(findViewById(R.id.content)) { view ->
            if (view !is TextView) return@setOnClickListenerForToggleButton
            Toast.makeText(this@MainActivity, "${view.text}", Toast.LENGTH_SHORT).show()
        }
    }
}